compiler/semstmts.nim

```diff
@@ -252,69 +252,61 @@ proc semCase(c: PContext, n: PNode): PNode =
     result.typ = typ
 
 proc semTry(c: PContext, n: PNode): PNode =
+
+  var check = initIntSet()
+  proc semExceptBranchType(typeNode: PNode): PNode =
+    let typ = semTypeNode(c, typeNode, nil).toObject()
+    if typ.kind != tyObject:
+      localError(typeNode.info, errExprCannotBeRaised)
+    if containsOrIncl(check, typ.id):
```

Comment:
Pass ``check`` explicitly to ``semExceptBranchType`` in order to avoid the closure creation.

Reply:
Closure creation removed by using template instead